packages/components/src/tabs/README.md

```diff
@@ -1,254 +1,216 @@
+`Tabs.Tablist` component and as many instances of the `Tabs.TabPanel`
+components as there are `Tabs.Tab` components.
 
--   Required: No
--   Default: `noop`
+ - Type: `ReactNode`
+ - Required: Yes
 
-###### `activeTabId`: `string | null`
+### `defaultTabId`
 
-The current active tab `id`. The active tab is the tab element within the tablist widget that has DOM focus.
+The id of the tab whose panel is currently visible.
 
-- `null` represents the tablist (ie. the base composite element). Users
-  will be able to navigate out of it using arrow keys;
-- If `activeTabId` is initially set to `null`, the base composite element
-  itself will have focus and users will be able to navigate to it using
-  arrow keys.
+If left `undefined`, it will be automatically set to the first enabled
+tab. If set to `null`, no tab will be selected, and the tablist will be
+tabbable.
 
-- Required: No
+Note: this prop will be overridden by the `selectedTabId` prop if it is
+provided (meaning the component will be used in "controlled" mode).
 
-###### `defaultActiveTabId`: `string | null`
+ - Type: `string`
+ - Required: No
 
-The tab id that should be active by default when the composite widget is rendered. If `null`, the tablist element itself will have focus and users will be able to navigate to it using arrow keys. If `undefined`, the first enabled item will be focused.
+### `defaultActiveTabId`
 
-_Note: this prop will be overridden by the `activeTabId` prop if it is provided._
+The tab id that should be active by default when the composite widget is
+rendered. If `null`, the tablist element itself will have focus
+and users will be able to navigate to it using arrow keys. If `undefined`,
+the first enabled item will be focused.
 
--   Required: No
+Note: this prop will be overridden by the `activeTabId` prop if it is
+provided.
 
-###### `onActiveTabIdChange`: `( ( activeId: string | null | undefined ) => void )`
+ - Type: `string`
+ - Required: No
+
+### `onSelect`
 
 The function called when the `selectedTabId` changes.
 
--   Required: No
--   Default: `noop`
+ - Type: `(selectedId: string) => void`
+ - Required: No
+
+### `onActiveTabIdChange`
 
-###### `orientation`: `'horizontal' | 'vertical' | 'both'`
+A callback that gets called when the `activeTabId` state changes.
 
-Defines the orientation of the tablist and determines which arrow keys can be used to move focus:
+ - Type: `(activeId: string) => void`
+ - Required: No
 
-- `both`: all arrow keys work;
-- `horizontal`: only left and right arrow keys work;
+### `orientation`
+
+Defines the orientation of the tablist and determines which arrow keys
+can be used to move focus:
+
+- `both`: all arrow keys work.
+- `horizontal`: only left and right arrow keys work.
 - `vertical`: only up and down arrow keys work.
 
--   Required: No
--   Default: `horizontal`
+ - Type: `"horizontal" | "vertical" | "both"`
+ - Required: No
+ - Default: `"horizontal"`
```

Comment:
The HTML rendering behavior of these "run-on" Markdown lists is not clearly defined, and depending on the renderer can result in a single merged list 😅

@ciampo Any ideas how to address systematically? (I usually just tweak the prop description so it doesn't end in a list item 🙈)

Reply:
Yes, sorry.

Let's say we have a prop description like this:

```ts
/**
 * This is my prop description. It has a list:
 *
 * - Foo
 * - Bar
 * - Baz
 */
myProp?: string;
```

That is going to end up in the README like this:

```md
### `myProp`

This is my prop description. It does three things:

- Foo
- Bar
- Baz

- Type: `string`
- Required: No
```

Paste that into a Markdown to HTML converter, and it's going to generate random weird HTML for that ambiguously separated list, depending on the converter implementation. There needs to be some content between those two lists for it to render reliably.

Reply:
Easiest thing to do that comes to mind is to change the format and stop having the type / required info displayed as a list. We could move them back on the same line as the prop name, maybe?

Reply:
Other simple solutions is to add a paragraph before the type/required to force an interruption of any previous lists

Reply:
> We could move them back on the same line as the prop name, maybe?

This ends up being a bit messy because some prop types can be quite long (like enums).

What do you think about #68301? I tried flipping the order, which should probably work better.